src/frontend/App.js

```diff
@@ -0,0 +1,51 @@
+import './stylesheets/style.css';
+import './stylesheets/global.css';
+
+import Router from './Router.js';
+import AccountHistoryDetailModel from './models/account-history/DetailModel.js';
+import AccountHistoryCalendarModel from './models/account-history/CalendarModel.js';
+
+import GlobalHeader from './components/GlobalHeader.js';
+import AccountHistoryDetailView from './views/account-history/DetailView.js';
+import AccountHistoryCalendarView from './views/account-history/CalendarView.js';
+
+const TAB_DATA = {
+  '/detail': {
+    Model: AccountHistoryDetailModel,
+    View: AccountHistoryDetailView,
+  },
+  '/calendar': {
+    Model: AccountHistoryCalendarModel,
+    View: AccountHistoryCalendarView,
+  },
+};
+
+export default class App {
+  constructor({$parent}) {
+    const $app = document.createElement('div');
+    $app.id = 'app';
+    this.$target = $app;
+    $parent.appendChild($app);
+
+    Router.subscribe('pathname', this.renderView.bind(this));
+    this.render();
+  }
+
+  render() {
+    new GlobalHeader();
+    this.renderView();
+  }
+
+  renderView() {
+    this.$view?.remove();
+
+    const pathname = Router.get('pathname');
+    if (pathname === '/statistics') return; // 나중에 statistics 추가하면 제거
+    const {Model, View} = TAB_DATA[pathname];
+
+    const model = new Model();
```

Comment:
```suggestion
    const {Model, View} = TAB_DATA[pathname];

    const model = new Model(pathname);
```

모델을 만들 때 `pathname`을 전해주고, 해당 모델이 사용되는 `pathname`을 알도록 `pathname`을 전달하고

Reply:
헉 제가 이 코멘트를 못 보고 위에 댓글을 달았었네요 ..ㅎㅎ
만약 이 방식을 사용하게 되면
Observer 클래스의 constructor 매개변수에 pathname을 추가하고 멤버 변수로 pathname을 추가하는 로직이 필요해서
Observer 클래스의 역할에 맞지 않는 코드가 추가된다고 생각해요..!!
그래서 개인적으로 Observer 클래스는 그 이름에 맞게 현재 코드를 유지하고
위에 제가 남긴 코멘트처럼 Model들 각각에서 처리하는 게 더 좋다고 생각합니다 !!

Reply:
넵!! 말씀 듣고 생각해보니 `pathname`이 `Observer` 클래스의 역할과 맞지 않는게 와닿네요 ㅎㅎ 
위에 남긴 코멘트처럼 처리하시면 좋을 것 같습니다!